### ISSUE TITLE
Encoding R2R issue for 10bit

### ISSUE THREAD

**Issue by spawlows** [OpenVisualCloud/SVT-AV1#1312]
Command line:
`-w 1920 -h 1080 -i Test_10bit_1920x1080_50Hz_P420.yuv  -b out.ivf -bit-depth 10 -n 100 -enc-mode 8`

I reverted 120 commits and issue still exist.
For param: -lp less or equal 3 issue does not exist.

**Comment by OmranAbdelkafi** [comment]
Hi @spawlows Could you please share the yuv used ? Thanks

**Comment by spawlows** [comment]
I use ParkJoy_1920x1080_10bit_unpacked_50Hz_P420.yuv to reproduce issue.

**Comment by EricLamVCD** [comment]
I tried to reproduce the R2R issue with the ParkJoy_1920x1080_10bit_unpacked_50Hz_P420.yuv clip with commit #326f320 but no r2r was found.

**Comment by silverbacknet** [comment]
@spawlows What CPU are you using? Or perhaps, what is the overall system config?

**Comment by spawlows** [comment]
Hi,

I tried to reproduce this on few platforms. On my laptop i can reproduce this every time, but on server i can not reproduce this too.

I found that this command reproduce issue on my laptop and server with windows:
```SvtAv1EncApp.exe -w 1920 -h 1080 -i ParkJoy_1920x1080_10bit_unpacked_50Hz_P420.yuv  -b PJ7.av1 -bit-depth 10 -n 100 -enc-mode 8 -lp 8```

**Comment by spawlows** [comment]
Hi,
I realized that ParkJoy_1920x1080_10bit_unpacked_50Hz_P420.yuv have unsupported 10bit format and this will reproduce issue on garbage video. 
For correct video which reproduce issue I am using Cactus_10bit_1920x1080_50Hz_P420.yuv

**Comment by OmranAbdelkafi** [comment]
Hi @spawlows for the case of ParkJoy_1920x1080_10bit_unpacked_50Hz_P420.yuv how many runs you will need to see the issue ?I still cannot see it? even if it s not a supported format , garbage video won't essentially cause a R2R.
Also what is your laptop CPU and specifications ? We are trying to reproduce the same environment.

**Comment by spawlows** [comment]
Hi,

I do tests on this commit: 326f320

My laptop is Intel(R) Core(TM) i5-8350U, 4 cores 8 threads. I agree that garbage video won't essentially cause a R2R. Usually it is  enough to run 2 or 3 times to see differences in bitstreams.
With parameter '-lp 8' (on my laptop it is default value because i have 8 threads in CPU) that same issue i can reproduce on 2 CPU platform: Intel(R) Xeon(R) Gold 6252. Without  '-lp 8' on 2 CPU platform it is impossible to reproduce for me. Both laptop and server are using windows 10 OS with Visual Studio 2017.

Please use command line: `SvtAv1EncApp.exe -w 1920 -h 1080 -i  Cactus_10bit_1920x1080_50Hz_P420.yuv -b out.av1 -bit-depth 10 -n 100 -enc-mode 8 -lp 8`

Output for this cmd return at least two different results
bistream md5sum:
243c7b8dc907442f5bd1f3ea042588e1 
e77cf378d3f94123980b4251f9e491e6

**Comment by ttrigui** [comment]
issue is reproducible with latest master [https://github.com/OpenVisualCloud/SVT-AV1/commit/ba72bc8511ed6a31544152590740e85f99a41300](url) 
However issue is NOT reproducible in svt-01[https://github.com/OpenVisualCloud/SVT-AV1/commit/3fdc26495cf88b2a17dcb5251a28bb2282c240d7](url)
The issue starts in frame 80 at the boundary CU (Cb recon different from r2r due to difference in residual)
After adopting the two bug fixes related to incomplete SB (INCOMPLETE_SB_FIX and FIX_INCOMPLETE_SB) from svt-01 the issue still persists in master.
@chkngit and @lijing0010 do you recall any bug-fix that was introduce that might fix this issue?

**Comment by 1480c1** [comment]
I'm currently seeing some r2r issues with some clips in objective-2-slow test set in 2d71125c966b6f16d3a675e3b7b15ee695a4cdb5 (master). I have not tested svt-01 yet

I am not sure if these are all of the problematic combinations.

version: `v0.8.3-65-g2d71125`

clips (preset qp, file):

- 8 43 objective-2-slow/Netflix_Boat_1920x1080_60fps_8bit_420_60f.y4m
- 8 20 objective-2-slow/street_hdr_amazon_2160p.y4m
- 4 63 objective-2-slow/Netflix_ToddlerFountain_4096x2160_60fps_10bit_420_60f.y4m
- 4 63 objective-2-slow/street_hdr_amazon_2160p.y4m
- 4 55 objective-2-slow/guitar_hdr_amazon_1080p.y4m
- 4 43 objective-2-slow/guitar_hdr_amazon_1080p.y4m
- 4 32 objective-2-slow/guitar_hdr_amazon_1080p.y4m
- 4 32 objective-2-slow/Netflix_BoxingPractice_4096x2160_60fps_10bit_420_60f.y4m
- 4 32 objective-2-slow/seaplane_hdr_amazon_1080p.y4m
- 4 20 objective-2-slow/guitar_hdr_amazon_1080p.y4m
- 4 20 objective-2-slow/Netflix_BarScene_4096x2160_60fps_10bit_420_60f.y4m
- 4 20 objective-2-slow/Netflix_BoxingPractice_4096x2160_60fps_10bit_420_60f.y4m
- 4 20 objective-2-slow/Netflix_WindAndNature_4096x2160_60fps_10bit_420_60f.y4m
- 4 20 objective-2-slow/street_hdr_amazon_2160p.y4m

commands used:

```bash
# --enable-hdr 1 is appended to those who have `_hdr_` in the name

./Bin/Release/SvtAv1EncApp -n 60 --keyint 63 --lookahead 0 --preset $preset -q $qp -i $file -b bitstreams/1-svt_M${preset}_${qp}_${file%.y4m}.ivf
./Bin/Release/SvtAv1EncApp -n 60 --keyint 63 --lookahead 0 --preset $preset -q $qp -i $file -b bitstreams/2-svt_M${preset}_${qp}_${file%.y4m}.ivf
diff bitstreams/1-svt_M${preset}_${qp}_${file%.y4m}.ivf bitstreams/2-svt_M${preset}_${qp}_${file%.y4m}.ivf
```

**Comment by chkngit** [comment]
we have just fixed few R2R issues in svt-01, can you give it a shot?

**Comment by 1480c1** [comment]
Sure, I can run the tests there

**Comment by chkngit** [comment]
help @hassount

**Comment by 1480c1** [comment]
No issues on svt-01

**Comment by tszumski** [comment]
I tried debug (with #1370) command line: ```SvtAv1EncApp.exe -w 1920 -h 1080 -i Cactus_10bit_1920x1080_50Hz_P420.yuv -b out.av1 -bit-depth 10 -n 100 -enc-mode 8 -lp 8```

I disabled all filtering and reduced number of frames to 82 (R2R still persist)
```
./SvtAv1EncApp.exe -i Cactus_10bit_1920x1080_50Hz_P420.yuv -w 1920 -h 1080 -bit-depth 10 -n 82 -enc-mode 8 -lp 8 --disable-dlf 1 --cdef-mode 0 --enable-restoration-filtering 0 --sg-filter-mode 0 --wn-filter-mode 0 --enable-intra-angle-delta 0 --enable-filter-intra 0 --enable-intra-edge-filter 0  -b out.ivf
```
The first diffrence is visible is frame 65, SB 501, at block position (1360,1072) of size 8x8 for chroma

**Comment by 1480c1** [comment]
@tszumski, when you ran the r2r tests on objective-2-slow, did you also run `street_hdr_amazon_2160p.y4m`?
On version v0.8.4-14-g73793b0 (73793b0), it seems to fail when I try to run
```bash
./Bin/Release/SvtAv1EncApp -n 60 --keyint 63 --lookahead 0 --preset 8 -q 20 -i objective-2-slow/street_hdr_amazon_2160p.y4m -b bitstreams/pr-svt_M8_20_street_hdr_amazon_2160p.ivf --enable-hdr 1
```

**Comment by 1480c1** [comment]
Thanks, it seems that maybe `--enable-hdr 1` doesn't actually work, so it would probably be a separate issue

**Comment by spawlows** [comment]
From my investigation R2R-10bit issue exists constantly from commit bc5a65894  PR #1040

@NaderMahdi can you helps us how to find reason of this issue in this changes? I know that some R2R issue for 8bit related with this commit was be fixed in the meantime.



Commit message:
commit bc5a658947a6abd9f8fa8240f80f5475fb91d87f
Author: NaderMahdi <43865616+NaderMahdi@users.noreply.github.com>
Date:   Wed Feb 19 13:27:50 2020 -0800

    Multi stages me (#1040)

    Multi stages me

    General Idea:
    Restructure the motion estimation to perform the following steps:
    1- HME search for all available N reference pictures
    2 - Select a subset M <= Nof reference pictures to be used for the next steps
    3- Integer-pel search on the N reference pictures
    4- Select a subset H <= M of reference pictures to be used for the next steps
    5- Subpel search on the H reference pictures

    Algorithm details:
    If HME SAD deviation to Best SAD is > TH%=80% (default) -> Prune reference MV ELSE IF
    If Fullpel ME SAD deviation to Best SAD is > TH%=30% (default) -> Prune reference MV

    Co-authored-by: chkngit <42250551+chkngit@users.noreply.github.com>

**Comment by hassount** [comment]
the macro PR_1311 seems to have different names between master and svt-01 such might have caused this issue

**Comment by hassount** [comment]
can we re-test thin in the latest master, it should be gone by now

**Comment by hassount** [comment]
Closing as this should be addressed after #1413
